src/ui/RealmBrowser/Content/CreateObjectDialog/index.tsx

```diff
@@ -85,18 +86,20 @@ class CreateObjectDialogContainer extends React.PureComponent<
     propertyName?: string,
     primaryKey?: string,
   ) {
-    // TODO: Initialize the values based on their property
-    if (
-      propertyName === 'uuid' &&
-      propertyName === primaryKey &&
-      property.type === 'string'
-    ) {
-      return uuid();
+    if (propertyName === primaryKey) {
+      // Special handling for primary keys. Opting out of optional handling & return a random value.
+      if (property.type === 'object id') {
+        return new ObjectId();
+      } else if (propertyName === 'uuid' && property.type === 'string') {
+        return uuid();
+      }
```

Comment:
I assume we need an else-if statement for all types that are possible as primary keys, in particular I think we need a branch for `int`. To avoid accidentally returning `undefined` I would suggest adding an `else` statement that throws with:

```typescript
throw new Error(`Unexpected type "${property.type}" as primary key`);
```

Reply:
True, update incoming. (separating primary key handling from the default if chain, so we fall back if no special-case returns).